### PR TITLE
drivers: i3c: add vendor and setaasa commands

### PR DESCRIPTION
**Pull request by XenuIsWatching**
Add CCC vendor and setaasa commands.

Also, some general fixes and enhancements.

### CODE REVIEW

File: drivers/i3c/i3c_ccc.c [hunk 1]
```diff
@@ -591,3 +591,110 @@ int i3c_ccc_do_getcaps(const struct i3c_device_desc *target,
...
 }
+
+int i3c_ccc_do_setvendor(struct i3c_device_desc *target, uint8_t id, uint8_t *payload, size_t len)
+{
+	struct i3c_ccc_payload ccc_payload;
+
+	__ASSERT_NO_MSG(target != NULL);
+
+	/* CCC must be between 0xE0 and 0xFE */
+	if (id > 30) {
```
Comment (dcpleung) [inline drivers/i3c/i3c_ccc.c:602]: Better use hex here since the comment is in hex.

File: drivers/i3c/i3c_ccc.c [hunk 2]
```diff
@@ -591,3 +591,110 @@ int i3c_ccc_do_getcaps(const struct i3c_device_desc *target,
...
+int i3c_ccc_do_setvendor(struct i3c_device_desc *target, uint8_t id, uint8_t *payload, size_t len)
+{
+	struct i3c_ccc_payload ccc_payload;
+
+	__ASSERT_NO_MSG(target != NULL);
+
+	/* CCC must be between 0xE0 and 0xFE */
+	if (id > 0x1E) {
+		return -EINVAL;
+	}
```
Comment (gmarull) [inline drivers/i3c/i3c_ccc.c:604]: shouldn't this check for < 0xE0 and > 0xFE according to the comment?

Comment (XenuIsWatching) [reply drivers/i3c/i3c_ccc.c:604]: no, `id` used with the macro `I3C_CCC_VENDOR`
```c
#define I3C_CCC_VENDOR(broadcast, id)		((id) + ((broadcast) ? 0x61U : 0xE0U))
```

**Review by dcpleung — DISMISSED** [review]

**Comment by XenuIsWatching** [comment]
rebasing to work around the regression introduced causing the CI to fail

File: drivers/i3c/i3c_shell.c [hunk 3]
```diff
@@ -1194,6 +1262,142 @@ static int cmd_i3c_ccc_getcaps(const struct shell *shell_ctx, size_t argc, char
...
+		shell_error(shell_ctx, "I3C: Device %s not attached to bus.", tdev->name);
+		return -ENODEV;
+	}
+	data = (struct i3c_driver_data *)dev->data;
+
+	id = strtol(argv[3], NULL, 16);
+	if (id > 0x1E) {
+		shell_error(shell_ctx, "I3C: CCC VENDOR ID %d is out of range.", id);
+		return -EINVAL;
+	}
```
Comment (gmarull) [inline drivers/i3c/i3c_shell.c:1349]: why is shell double-checking here? I3C code will already fail.

Comment (XenuIsWatching) [reply drivers/i3c/i3c_shell.c:1349]: it's to give a specific error print back to the user

Comment (gmarull) [reply drivers/i3c/i3c_shell.c:1349]: I don't think this is a good error handling practice, `-EINVAL` can be documented as a value returned for that condition. It's true that it is difficult to be precise just with errno, but that's what we have... Logging in the driver can also help (if activated, user will get more precise info)

Comment (XenuIsWatching) [reply drivers/i3c/i3c_shell.c]: removed

**Review by dcpleung — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**